### PR TITLE
feat:  change dataset uri to return full qualified url instead of object store path

### PR DESCRIPTION
**Pull request by eddyxu**
BREAKING CHANGE

### CODE REVIEW

File: rust/lance-index/src/vector/hnsw/builder.rs [hunk 1]
```diff
@@ -790,7 +791,7 @@ mod tests {
         .unwrap();
 
         let object_store = ObjectStore::memory();
-        let path = object_store.base_path().child("test_builder_write_load");
+        let path = Path::from("test_builder_write_load");
```
Comment (chebbyChefNEQ) [inline rust/lance-index/src/vector/hnsw/builder.rs:794]: why this change?

File: rust/lance-io/src/object_store.rs [hunk 2]
```diff
@@ -707,14 +696,6 @@ async fn configure_store(url: &str, options: ObjectStoreParams) -> Result<Object
         "s3" | "s3+ddb" => {
             storage_options.with_env_s3();
 
-            // if url.scheme() == "s3+ddb" && options.commit_handler.is_some() {
```
Comment (chebbyChefNEQ) [inline rust/lance-io/src/object_store.rs:710]: I think we should reenable this instead of delete it

File: rust/lance-index/src/vector/hnsw/builder.rs (hunk 1 above)
Comment (eddyxu) [reply rust/lance-index/src/vector/hnsw/builder.rs:794]: `ObjectStore::base_path` carries the same `base path` as `Dataset::base`. Logically, "base path" is not a attribute to `ObjectStore` which should be an handler for IO?

File: rust/lance-io/src/object_store.rs (hunk 2 above)
Comment (eddyxu) [reply rust/lance-io/src/object_store.rs:710]: Sure , will do

**Review by chebbyChefNEQ — APPROVED** [review]

File: rust/lance/src/dataset.rs [hunk 3]
```diff
@@ -590,9 +599,9 @@ impl Dataset {
         self.append_impl(batches, params).await
     }
 
-    /// Get the base URI of the dataset.
-    pub fn uri(&self) -> &Path {
-        &self.base
+    /// Get the fully qualified URI of this dataset.
+    pub fn uri(&self) -> &str {
+        &self.uri
     }
```
Comment (wjones127) [inline rust/lance/src/dataset.rs:605]: This is a breaking change to a public API. I've labelled it as such.

However, if we didn't want to make a breaking change, you could instead make a new method and deprecate this one.

Comment (eddyxu) [reply rust/lance/src/dataset.rs:605]: Let's bump the version. The rust API does not have a large audience yet
https://crates.io/crates/lance/reverse_dependencies

This new release process does not allow us to accumulate a few breaking changes together for a new version? 
https://github.com/lancedb/lance/blob/main/release_process.md#breaking-change-policy this is suggesting i manually bump the version, right?

Comment (wjones127) [reply rust/lance/src/dataset.rs:605]: > This new release process does not allow us to accumulate a few breaking changes together for a new version?

You can accumulate them. Once you've bumped the version once, you don't have to do it again until the next release.
